pandas/_libs/tslibs/np_datetime.pyx

```diff
@@ -147,6 +147,9 @@ cdef inline void td64_to_tdstruct(int64_t td64,
 
 cdef inline int64_t pydatetime_to_dt64(datetime val,
                                        npy_datetimestruct *dts):
+    """
```

Comment:
add an explicit check?

Reply:
This is called by some very perf-sensitive code, all uses of which already handle this carefully.  I think its sufficiently internal to be OK without.